### PR TITLE
VRF Lite Support - adding route

### PR DESCRIPTION
**Pull request by HenocKA**
Discovery route for get all ip tables from cisco (ospf, bgp, direct...)

### CODE REVIEW

**Comment by laf** [comment]
This needs a rebase (most likely the schema change). 

I'd say the module should default to 0 so it's off, having this on could add substantial time to discovery for people who run a lot of Cisco kit.

You will also need to update:

doc/Support/Discovery\ Support.md

to include this new module.

**Comment by jpasquie** [comment]
So, we changing the default to 0 for enable_route.
We see for the base mapping.
And after we change the doc.

**Comment by laf** [comment]
Unless anyone ( @librenms/reviewers ) disagree then yes. This is something that will take some time, especially on busy routers so to have it enabled by default seems ott to me.

File: includes/defaults.inc.php [hunk 1]
```diff
@@ -728,6 +730,7 @@ function set_debug($debug) {
 $config['discovery_modules']['cisco-vrf-lite']       = 1;
 $config['discovery_modules']['ipv4-addresses']       = 1;
 $config['discovery_modules']['ipv6-addresses']       = 1;
+$config['discovery_modules']['route']                = 1;
```
Comment (laf) [inline includes/defaults.inc.php]: This needs to be disabled by default.

File: includes/defaults.inc.php [hunk 2]
```diff
@@ -440,6 +440,8 @@ function set_debug($debug) {
 // Enable VRFs
 $config['enable_vrf_lite_cisco'] = 1;
 // Enable VRF lite cisco
+$config['enable_route'] = 0;
```
Comment (laf) [inline includes/defaults.inc.php]: Please remove this, with it, it means people have to enable two config options when $config['discovery_modules']['route'] does the same thing.

**Comment by laf** [comment]
I've just commented on a couple of things and also as with the other PR it could do with being tidied up. You have a number of commits and file changes unrelated to this one.

**Comment by Tristan-Labbe** [comment]
I changed the file to include your comments.

**Comment by laf** [comment]
It just looks like you've removed the config option and not the code that checks it's enabled so that means the discovery would never run.

The PR really needs to be tidied up as well, this includes a lot of changes that are already in the master branch

**Comment by laf** [comment]
Thanks for tidying it up :)

However, you still have if ($config['enable_route']) { in your discovery file.

**Comment by Tristan-Labbe** [comment]
Yes. 
However without "$config['enable_route']", the discovery will be always activate.

**Comment by laf** [comment]
You're evaluating $config['enable_route'] as true in that if statement so the discovery won't be run. However as I've mentioned before, you shouldn't have two config options to enable this, the discovery module config is enough.

**Comment by Tristan-Labbe** [comment]
Ok. I change the condition. 
I didn't think that i could use  "$config['discovery_modules']['route']" here.

**Comment by laf** [comment]
I'm not being clear :/

You can drop if ($config['discovery_modules']['route']) { completely. The check if the module should run is done before it's even including your script so forget about that bit.

**Comment by laf** [comment]
All looks good to me on this. It's not dependant on anything else is it as it does a check to see if the vrf lite column exists before doing anything?

**Comment by Tristan-Labbe** [comment]
It does check to know the context name of the device.
If the device has not VRF, the context name is set to null.
It is only for the name of the device concerned .

**Comment by laf** [comment]
Ok, happy to merge once other PRs with the schema numbers before this are ready.

Feel free to change this to 113.sql if you want it merging sooner.

**Comment by Tristan-Labbe** [comment]
Bon boulot @HenocKA  :)
Je te laisse fermer la branche ?

**Pull request merged**